backend-rust/src/graphql_api/baker.rs

```diff
@@ -1045,6 +1076,519 @@ impl Baker {
         }
         Ok(connection)
     }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn block_commission_desc_connection(
+        config: &ApiServiceConfig,
+        pool: &PgPool,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+        open_status_filter: Option<BakerPoolOpenStatus>,
+        include_removed_filter: bool,
+    ) -> ApiResult<connection::Connection<String, Baker>> {
+        type RemovedBakerCursor = Reversed<BakerIdCursor>;
+        type Cursor = ConcatCursor<BakerFieldDescCursor, RemovedBakerCursor>;
+
+        /// Internal helper function for querying the current bakers sorted
+        /// by the pool_delegator_count in descending order.
+        async fn query_current_bakers(
+            query: ConnectionQuery<BakerFieldDescCursor>,
+            open_status_filter: Option<BakerPoolOpenStatus>,
+            connection: &mut connection::Connection<String, Baker>,
+            pool: &PgPool,
+        ) -> ApiResult<()> {
+            let mut row_stream = sqlx::query_as!(
+                CurrentBaker,
+                r#"SELECT * FROM (
+                SELECT
+                    bakers.id AS id,
+                    staked,
+                    restake_earnings,
+                    open_status as "open_status: _",
+                    metadata_url,
+                    transaction_commission,
+                    baking_commission,
+                    finalization_commission,
+                    payday_transaction_commission as "payday_transaction_commission?",
+                    payday_baking_commission as "payday_baking_commission?",
+                    payday_finalization_commission as "payday_finalization_commission?",
+                    payday_lottery_power as "payday_lottery_power?",
+                    payday_ranking_by_lottery_powers as "payday_ranking_by_lottery_powers?",
+                    (SELECT MAX(payday_ranking_by_lottery_powers) FROM bakers_payday_lottery_powers) as "payday_total_ranking_by_lottery_powers?",
+                    pool_total_staked,
+                    pool_delegator_count
```

Comment:
In general it is not good practice that we keep on spamming this select into so many different calls. This make altercations on bakers cumbersome and prone to errors. Can we define one place what a baker look like when we have this many calls?

Reply:
I don't think it is prone to errors due to the compile time checked queries.

But I agree that it is cumbersome. One solution would be to not use compile time checked queries, so we split up and compose the queries, but I think this becomes prone to errors, and we would need more testing to catch mistakes, which is probably more cumbersome.

Another solution would be to only fetch `id`s in this query and then have another query, which fetches from a list of `id`s, meaning we only list every selected field in one query. Here it would mean a hit in performance instead, but I am not sure how much though and whether it would be worth it.

Reply:
We can consider it another task?

Reply:
I am writing a task on it. It could also be that sqlx defines a way in how to map from a Struct to a select statement.

Reply:
https://linear.app/concordium/issue/CCD-207/unify-selects-on-bakers